src/jit/lsraarm.cpp

```diff
@@ -664,13 +664,10 @@ void LinearScan::TreeNodeInfoInit(GenTree* tree)
         break;
 
         case GT_NULLCHECK:
-            // Although NULLCHECK is defined as GTK_NOVALUE, it requires a target
-            // register on ARM, as it is implemented as a load.
-            info->dstCount      = 1;
-            info->srcCount      = 1;
-            info->isLocalDefUse = true;
-            // null check is an indirection on an addr
-            TreeNodeInfoInitIndir(tree->AsIndir());
+            // It requires a internal register on ARM, as it is implemented as a load
+            assert(info->dstCount == 0);
+            info->srcCount         = 1;
+            info->internalIntCount = 1;
             break;
```

Comment:
I'm not sure but how about leave ` TreeNodeInfoInitIndir(tree->AsIndir());` as it ?

Reply:
`TreeNodeInfoInitIndir()` set internal register by checking child node. But we want only one internal register on ARM32 and no internal register on ARM64 for `GT_NULLCHECK`. So it seems useless call.

Reply:
If you are not calling `TreeNodeInfoInitIndir()` here or in lsraarm64.cpp, then you should add an assert that the address is not contained.

Reply:
Actually, I see that we don't call `TryCreateAddrMode()` for `GT_NULLCHECK` (presumably because on x86/x64 it's not actually implemented as a load), but I still think you should add an assert.